From 478414c0db15d7264068092744436f17b9b5c9d6 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 20 Feb 2024 18:47:04 +0100
Subject: [PATCH 612/841] fix(stm32mp2): skip iac_dump when M33 is TDCID

When Cortex-M33 is TDCID, it has the control of IAC peripheral. Cortex-A
cannot access it. Put the code of iac_dump() under STM32MP_M33_TDCID
flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6be48d44a5fc8a87914c2967181eeb184fed7973 fix(stm32mp2): skip iac_dump when M33 is TDCID]
Change-Id: I3c020f55d1d61261491d870080c091439eb8e9b6
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/361645
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/385593
---
 plat/st/stm32mp2/bl2_plat_setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index cd66b2dda..0a1067f4e 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -55,6 +55,7 @@ IMPORT_SYM(uintptr_t, __DATA_END__, DATA_END);
 
 static void iac_dump(void)
 {
+#if !STM32MP_M33_TDCID
 #if DEBUG
 	unsigned int i;
 	unsigned int bit;
@@ -80,6 +81,7 @@ static void iac_dump(void)
 		}
 	}
 #endif
+#endif
 }
 
 static void print_reset_reason(void)
-- 
2.47.2

